Feature/view recipe list

orria98 opened the pull request:
## Það sem er hægt að gera:

* Sjá recipe listana sína ef maður er innskráður
* Ýta á recipe lista til að skoða nánar
* Ýta á recipe í listanum til að skoða það nánar

Megið endilega testa þetta og láta mig vita ef þið finnið eitthvað að, eins ef þið finnið eitthvað athugavert í kóðanum :)

orria98 committed:
View recipe list komið

orria98 committed:
Merge branch 'master' into feature/view_recipe_list

Sigurbj02 commented on app/src/main/java/hbv601g/recipeapp/adapters/RecipeListAdapter.java:
```diff
@@ -10,12 +11,14 @@
 import java.util.List;
 
 import hbv601g.recipeapp.R;
+import hbv601g.recipeapp.entities.Recipe;
```
Það má taka import skipanirnar sem þú bættir við út, þær eru ekkert notaðar

Sigurbj02 commented on app/src/main/java/hbv601g/recipeapp/entities/RecipeList.java:
```diff
@@ -12,7 +17,7 @@
  */
 
-public class RecipeList {
+public class RecipeList implements Parcelable {
```
Það kemur villa hér
![image](https://github.com/user-attachments/assets/eb3a01c6-7e37-465f-9c5c-bf64ac2723a6)

Sigurbj02 commented on app/src/main/java/hbv601g/recipeapp/service/RecipeListService.java:
```diff
@@ -20,7 +20,7 @@
 import java.util.List;
 
-
+import hbv601g.recipeapp.entities.Recipe;
```
Það má taka þetta út

Sigurbj02 commented on app/src/main/res/layout/fragment_recipe_list.xml:
```diff
@@ -0,0 +1,75 @@
+    tools:context=".ui.recipeLists.RecipeListFragment">
+
+    <TextView
+        android:id="@+id/recipe_list_title"
```
Ég myndi hafa titilinn á uppskriftinni stærri en "Description" section titillinn, annars er ég mjög hrifin af þessu

Sigurbj02 commented on app/src/main/java/hbv601g/recipeapp/ui/recipeLists/RecipeListFragment.java:
```diff
@@ -0,0 +1,85 @@
+import hbv601g.recipeapp.service.RecipeListService;
+
+
+public class RecipeListFragment extends Fragment {
```
Ef maður opnar recipe list og fer svo í recipe flipann og bætir uppskrift við listann þá uppfærist listinn ekki, heldur er hann settur sem argumentið í hvert skipti

Sigurbj02 reviewed (changes requested):
Ég setti nokkur comment sem má endilega skoða. Ég tók eftir því að ef maður er með recipeList opinn, fer svo í recipe flipann og bætir uppskrift við listann, þá uppfærist listinn ekki. Verðin á uppskriftunum eru líka vitlaus

orria98 committed:
Import skipanir teknar út
Bæta við aðferðum í RecipeList.java
Ef uppskrift er bætt við lista á meðan hann er opinn uppfærist hann sjálfkrafa
Uppskriftir úr lista sóttar frá API til að fá TIC og TPC

Sigurbj02 reviewed (commented):
Það vantar skjölun í RecipeListService og RecipeListFragment, en annars mjög gott

orria98 committed:
Documentation
